pkg/intermediate/aggregate.go

```diff
@@ -105,12 +105,16 @@ func (a *AggregationProcess) ForAllRecordsDo(callback FlowKeyRecordMapCallBack)
 	return nil
 }
 
-func (a *AggregationProcess) DeleteFlowKeyFromMap(flowKey FlowKey) {
+func (a *AggregationProcess) DeleteFlowKeyFromMapWithLock(flowKey FlowKey) {
 	a.flowKeyRecordLock.Lock()
 	defer a.flowKeyRecordLock.Unlock()
 	delete(a.flowKeyRecordMap, flowKey)
 }
 
+func (a *AggregationProcess) DeleteFlowKeyFromMapWithoutLock(flowKey FlowKey) {
+	delete(a.flowKeyRecordMap, flowKey)
+}
```

Comment:
how would users acquire the lock if they want to use this function? Should we make lock public?

Reply:
Added a comment. Ideally, we should not expose this. This is a temporary addition until we support the concept of invalid flows in Flow Exporter and thereby in the aggregation process, which will make flowKey deletion easier in flow aggregator of Antrea.